docs/docs/spark-configuration.md

```diff
@@ -145,6 +145,61 @@ Using those SQL commands requires adding Iceberg extensions to your Spark enviro
+| spark.wap.id                                           | null                                                           | [Write-Audit-Publish](branching.md#audit-branch) snapshot staging ID   |
+| spark.wap.branch                                       | null                                                           | WAP branch name for snapshot commit                                    |
+| spark.sql.iceberg.compression-codec                    | Table default                                                  | Write compression codec (e.g., `zstd`, `snappy`)                       |
+| spark.sql.iceberg.compression-level                    | Table default                                                  | Compression level for Parquet/Avro                                     |
+| spark.sql.iceberg.compression-strategy                 | Table default                                                  | Compression strategy (for ORC)                                         |
+| spark.sql.iceberg.data-planning-mode                   | Table default                                                  | Override for data planning mode                                        |
+| spark.sql.iceberg.delete-planning-mode                 | Table default                                                  | Override for delete planning mode                                      |
+| spark.sql.iceberg.advisory-partition-size              | Table default                                                  | Advisory size (bytes) for data planning                                |
+| spark.sql.iceberg.locality.enabled                     | false                                                          | Report locality information for task planning                          |
+| spark.sql.iceberg.executor-cache.enabled               | true                                                           | Enables cache for executor-side metadata                               |
+| spark.sql.iceberg.executor-cache.timeout               | 10 minutes                                                     | Timeout for executor cache entries                                     |
+| spark.sql.iceberg.executor-cache.max-entry-size        | 64MB                                                           | Max size per cache entry                                               |
+| spark.sql.iceberg.executor-cache.max-total-size        | 128MB                                                          | Max total executor cache size                                          |
+| spark.sql.iceberg.executor-cache.locality.enabled      | false                                                          | Enables locality-aware executor cache usage                            |
+| spark.sql.iceberg.merge-schema                         | false                                                          | Enables automatic schema merge during write                            |
```

Comment:
Enables modifying the table schema to match the write schema. Only adds columns missing columns.